ExeterTeam25/group-software-project
# Photo security and compression

To test profile pic compression and naming:
- Run the server and log in
- Go to profile and set a picture
- Go to the UniExplore\media\profile_pictures directory
- Ensure that the image has been uploaded with the convention `{userid}.jpg`, and the resolution and quality have been altered
- Set a new profile pic
- Ensure the new image has replaced the original file in the UniExplore\media\profile_pictures directory

To test response pic compression and naming:
- Run the server and log in
- Go to complete a challenge screen and submit a response
- Go to the UniExplore\media\image_uploads directory
- Ensure that the image has been uploaded with the convention `{userid}-{challengeid}.jpg`, and the resolution and quality have been altered

To test settings:
- Go to settings.py
- Mess around with the photo upload settings ( `RESPONSE_PHOTO_SIZE`, `RESPONSE_PHOTO_QUALITY`, `PROFILE_PIC_SIZE`, `PROFILE_PIC_QUALITY`
- Upload corresponding photos
- Ensure changes are reflected in the final files

Details:
Implemented uploaded images naming schemes
- Profile pic: `{userid}.jpg`
- Response photo: `{userid}-{challengeid}.jpg`
Added image resolution and compression setting on upload
Added image quality settings in config file
